### DISCUSSION TITLE
Goals

### DISCUSSION THREAD

**Discussion by frustak** [frustak/koal#13] [Ideas]
### Problem
I want to add goals

### Boundaries
goals does not need to have to-dos at first,
we can add them later in the planning page
support add, delete

### Rough Idea
at the home page, we show the list of goals.
i can add a new goal by clicking the add new goal button.
an input will be shown which I can type the goal name.
after adding the goal, goal list will be updated.
a delete button exists on the right side of goal.
adding new tasks to goals are handled in planning page.

![goals](https://user-images.githubusercontent.com/47036679/205646703-1703a218-0d30-488d-ac8f-6a60800bee69.png)

**Comment by Glyphack** [comment]
We have the APIs to create goals and todos. Let me know if anything did not work @frustak .

**Comment by frustak** [comment]
@Glyphack I've added a simple validation for goal name in UI
```ts
z.object({
  name: z.string().min(1).max(100),
});
```
